### DISCUSSION TITLE
How a Kong Basic Auth Plugin can be restricted to set of consumers?

### DISCUSSION THREAD

**Discussion by sasahoo-tibco** [Kong/kong#13374] [Help]
### Discussed in https://github.com/Kong/kong/discussions/13367

<div type='discussions-op-text'>

<sup>Originally posted by **sasahoo-tibco** July 12, 2024</sup>
I have set up the basic auth Kong Plugin and created a Consumer using a Kubernetes secret. This configuration works for all consumers using that secret. I have an ingress that uses this plugin. However, I want the Kong Plugin to work only with a specific set of consumers, not all. Specifically, I would like to know if the Kong Plugin can be configured to work with a single consumer.

For example:

Ingress 'A' uses Kong Basic Auth Plugin 'Ka' and should work only with Consumer 'A'.
Ingress 'B' uses Kong Basic Auth Plugin 'Kb' and should work only with Consumer 'B'.
Ingress 'A' should not authenticate Consumer 'B', and Ingress 'B' should not authenticate Consumer 'A'.

Is there a solution for this scenario?</div>

**Comment by hanshuebner** [comment]
Have a look here: https://discuss.konghq.com/t/limit-routes-to-specific-consumers/9774

Does this answer your question?

**Reply by sasahoo-tibco** [reply]
Thanks @hanshuebner. Do you have sample of ACL plugin for kubernetes I can refer?  It will be a great help if you add more info for ACL plugin.
